pkg/config/expand.go

```diff
@@ -467,20 +467,18 @@ func identifyModuleByReference(yamlReference string, dg DeploymentGroup) (modRef
 /*
 A variable reference has the following fields
   - ID: a module ID or "vars" if referring to a deployment variable
-  - GroupID: if ID is a module ID, GroupID must be the deployment group in
-    which the module is *expected* to be found. If ID is "vars", then it
-    should be set to "deployment" to indicate that the reference belongs
-    to the entire blueprint, rather than a deployment group.
   - Name: the name of the module output or deployment variable
-  - ExplicitInterGroup: a boolean value indicating whether the user made a
-    reference that superficially appears to be intergroup (i.e. they used
-    an explicit GroupID that differs from the context's GroupID)
+  - ToGroupID: the deployment group in which the module is *expected* to be found
+  - FromGroupID: the deployment group from which the reference is made
+  - Explicit: a boolean value indicating whether the user made a reference that
+    explicitly identified ToGroupID rather than inferring it using FromGroupID
 */
 type varReference struct {
-	ID                 string
-	GroupID            string
-	Name               string
-	ExplicitInterGroup bool
+	ID          string
```

Comment:
The problem with this is that, in the model, `ref.ID` can also be "vars" which is not a module. It might also complicate your suggestion at https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/858#discussion_r1087170425. I'd rather not address this issue at the time. The meaning is documented which is an improvement.